pkg/osinfo/container_scanner.go

```diff
@@ -108,6 +114,40 @@ func (ct *ContainerScanner) ReadDebianPackages(layers []string) (layer int, pk *
 	return layer, pk, err
 }
 
+// ReadApkPackages reads the last known changed copy of the apk database
+func (ct *ContainerScanner) ReadApkPackages(layers []string) (layer int, pk *[]PackageDBEntry, err error) {
+	apkDatabase := ""
+	loss := LayerScanner{}
+	for i, lp := range layers {
+		tmpDB, err := os.CreateTemp("", "apkdb-")
+		if err != nil {
+			return 0, pk, fmt.Errorf("opening temporary apkdb file: %w", err)
+		}
+		tmpDBPath := tmpDB.Name()
+		defer os.Remove(tmpDBPath)
```

Comment:
Defer in a loop not recommended: https://github.com/go-critic/go-critic/blob/2c6e174eb586a4e13d07c39cbff77fb43c7aff6e/docs/overview.md#deferinloop

Reply:
Thanks @saschagrunert ! Sorry I missed your review and had forgotten about this PR. I removed the defer in this commit and also removed another one in the debian parser. I also pushed another commit to fix another small bug.

PTAL :")